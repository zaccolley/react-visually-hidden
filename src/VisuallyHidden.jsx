import React from "react";
import PropTypes from "prop-types";
import { getTextFromChildren } from "./helpers";
import styles from "./styles.css";

const VisuallyHidden = ({ children }) => {
  if (!children) {
    return null;
  }

  const content = getTextFromChildren(children);

  if (!content) {
    return null;
  }

  return <span className={styles.visuallyHidden}>{content}</span>;
};

VisuallyHidden.propTypes = {
  children: PropTypes.node
};

VisuallyHidden.defaultProps = {
  children: null
};

export default VisuallyHidden;
