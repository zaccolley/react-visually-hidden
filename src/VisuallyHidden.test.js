import React from "react";
import renderer from "react-test-renderer";
import VisuallyHidden from "./VisuallyHidden";

it("should render null for nothing sent", () => {
  const component = renderer.create(<VisuallyHidden />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("should just render text", () => {
  const component = renderer.create(
    <VisuallyHidden>
      Hello this is text and should look like text
    </VisuallyHidden>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("should just render the text from various JSX elements", () => {
  const component = renderer.create(
    <VisuallyHidden>
      <h1>Hello</h1>{" "}
      <p>
        This is text and should <strong>look</strong> like text
      </p>
    </VisuallyHidden>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
