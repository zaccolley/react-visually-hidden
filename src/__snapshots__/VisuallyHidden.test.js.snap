// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should just render text 1`] = `
<span
  className="visuallyHidden"
>
  Hello this is text and should look like text
</span>
`;

exports[`should just render the text from various JSX elements 1`] = `
<span
  className="visuallyHidden"
>
  Hello This is text and should look like text
</span>
`;

exports[`should render null for nothing sent 1`] = `null`;
